components/brave_wallet/browser/asset_discovery_manager.cc

```diff
@@ -48,16 +53,129 @@ AssetDiscoveryManager::GetAssetDiscoverySupportedChains() {
   return *asset_discovery_supported_chains;
 }
 
-void AssetDiscoveryManager::DiscoverAssets(
+void AssetDiscoveryManager::DiscoverSolAssets(
+    const std::vector<std::string>& account_addresses,
+    bool triggered_by_accounts_added) {
+  if (account_addresses.empty()) {
+    CompleteDiscoverAssets(
+        mojom::kSolanaMainnet, std::vector<mojom::BlockchainTokenPtr>(),
+        absl::nullopt, l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS),
+        triggered_by_accounts_added);
+    return;
+  }
+
+  // TODO(nvonpentz): When custom networks are supported, we need to check
+  // that the active network is our own that supports this RPC call.
+
+  const auto barrier_callback = base::BarrierCallback<std::vector<std::string>>(
+      account_addresses.size(),
+      base::BindOnce(&AssetDiscoveryManager::MergeDiscoveredSolanaAssets,
+                     weak_ptr_factory_.GetWeakPtr(),
+                     triggered_by_accounts_added));
+  for (const auto& account_address : account_addresses) {
```

Comment:
Should we limit to a fixed number of addresses? Thinking of edge cases where the wallet might have thousands Solana accounts.

Reply:
I'll need to go back and find the conversation, but I remember having a conversation with @darkdh about how there was an optimized way to limit the number of accounts we checked. It may have just been a naive solution like only check upto a list of 100, but I can't remember. Let me dig a bit more and get back to you on this for a recommendation because I agree that we don't want to be checking 100s or 1000s of accounts.